Lib/traceback.py

```diff
@@ -912,34 +914,67 @@ def format_exception_only(self, *, show_group=False, _depth=0):
                 smod = "<unknown>"
             stype = smod + '.' + stype
 
-        if not issubclass(self.exc_type, SyntaxError):
-            if _depth > 0:
-                # Nested exceptions needs correct handling of multiline messages.
-                formatted = _format_final_exc_line(
-                    stype, self._str, insert_final_newline=False,
-                ).split('\n')
-                yield from [
-                    indent + l + '\n'
-                    for l in formatted
-                ]
-            else:
-                yield _format_final_exc_line(stype, self._str)
+        if issubclass(self.exc_type, BaseExceptionGroup):
+            limit = self.max_group_depth - 1  # minus ExceptionGroup itself
         else:
-            yield from [indent + l for l in self._format_syntax_error(stype)]
-
-        if (
-            isinstance(self.__notes__, collections.abc.Sequence)
-            and not isinstance(self.__notes__, (str, bytes))
-        ):
-            for note in self.__notes__:
-                note = _safe_string(note, 'note')
-                yield from [indent + l + '\n' for l in note.split('\n')]
-        elif self.__notes__ is not None:
-            yield indent + "{}\n".format(_safe_string(self.__notes__, '__notes__', func=repr))
+            limit = self.max_group_depth
```

Comment:
it seems that there is a lot of repetition now between format and format_exception_only. Could we not have format take an arg to makes it skip the traceback (so that it becomes format_exception_only)?

Reply:
I will take a look 👍